src/autocat/learning/sequential.py

```diff
@@ -172,50 +168,379 @@ def update(self, structures: List[Atoms], labels: Array):
                         self.design_space_labels, labels[i]
                     )
 
-    def to_jsonified_list(self) -> List:
+    def to_jsonified_dict(self) -> Dict:
         """
-        Returns a jsonified list representation
+        Returns a jsonified dict representation
         """
-        collected_jsons = []
+        collected_structs = []
         for struct in self.design_space_structures:
-            collected_jsons.append(atoms_encoder(struct))
-        # append labels to list of collected jsons
+            collected_structs.append(atoms_encoder(struct))
         jsonified_labels = [float(x) for x in self.design_space_labels]
-        collected_jsons.append(jsonified_labels)
-        return collected_jsons
+        return {"structures": collected_structs, "labels": jsonified_labels}
 
     def write_json_to_disk(
-        self,
-        json_name: str = None,
-        write_location: str = ".",
-        write_to_disk: bool = True,
+        self, json_name: str = None, write_location: str = ".",
     ):
         """
         Writes DesignSpace to disk as a json
         """
-        collected_jsons = self.to_jsonified_list()
+        collected_jsons = self.to_jsonified_dict()
         # set default json name if needed
         if json_name is None:
             json_name = "acds.json"
-        # write out single json
-        if write_to_disk:
-            json_path = os.path.join(write_location, json_name)
-            with open(json_path, "w") as f:
-                json.dump(collected_jsons, f)
+
+        json_path = os.path.join(write_location, json_name)
+        with open(json_path, "w") as f:
+            json.dump(collected_jsons, f)
 
     @staticmethod
-    def from_json(json_name: str):
-        with open(json_name, "r") as f:
-            all_data = json.load(f)
+    def from_jsonified_dict(all_data: Dict):
         structures = []
-        for i in range(len(all_data) - 1):
-            atoms = atoms_decoder(all_data[i])
+        for i in range(len(all_data["structures"])):
+            atoms = atoms_decoder(all_data["structures"][i])
             structures.append(atoms)
-        labels = np.array(all_data[-1])
+        labels = np.array(all_data["labels"])
         return DesignSpace(
             design_space_structures=structures, design_space_labels=labels,
         )
 
+    @staticmethod
+    def from_json(json_name: str):
+        with open(json_name, "r") as f:
+            all_data = json.load(f)
+        return DesignSpace.from_jsonified_dict(all_data)
+
+
+class CandidateSelectorError(Exception):
+    pass
+
+
+class CandidateSelector:
+    def __init__(
+        self,
+        acquisition_function: str = None,
+        num_candidates_to_pick: int = None,
+        target_window: Array = None,
+        include_hhi: bool = None,
+        hhi_type: str = "production",
+        include_segregation_energies: bool = None,
```

Comment:
Similar to `hhi_type`, aren't there two sources of `segregation_energies` (`rao2020`, etc.) the user can choose from? Should we add another parameter here similar to `hhi_type` for segregation energies?